src/reactjs/logger.native.ts

```diff
@@ -1,7 +0,0 @@
-import type { Logger } from '../core/logger'
-
-export const logger: Logger = {
-  log: console.log,
-  warn: console.warn,
-  error: console.warn,
```

Comment:
Was this a mistake or was it on purpose to sure `console.warn` for `logger.error` for react-native?

Reply:
Ok, I digged out https://github.com/tannerlinsley/react-query/issues/795 so it's undesirable to use `console.error` for react-native users and that was the primary reason for `logger.native.ts` I assume. 

I'll change it.

Reply:
Or should we ask react-native users to pass a custom logger to `QueryClient` instead?

Reply:
good question. @arnaudbzn ?

Reply:
The current default logger behavior for React Native is fine. 
We should not require React Native developers to define a custom logger.